enterprise/backend/src/metabase_enterprise/serialization/v2/extract.clj

```diff
@@ -62,21 +62,16 @@
     (eduction (map #(serdes/extract-all % extract-opts)) cat (model-set opts))))
 
 ;; TODO Properly support "continue" - it should be contagious. Eg. a Dashboard with an illegal Card gets excluded too.
-(defn- chase-for [targets func]
-  (loop [to-chase (set targets)
-         chased   #{}]
-    (let [[m i :as item] (first to-chase)
-          res            (func m i)
-          chased         (conj chased item)
-          to-chase       (set/union (disj to-chase item) (set/difference res chased))]
-      (if (empty? to-chase)
-        chased
-        (recur to-chase chased)))))
-
-(defn- gather-lineage [targets]
-  (set/union
-   (chase-for targets serdes/ascendants)
-   (chase-for targets serdes/descendants)))
+(defn- traverse-with [targets traverse-fn]
```

Comment:
Last idea - now that `traverse-with` is abstract and has less to do with serialization, we should rename `targets` to `nodes` to be consistent